src/views/List.jsx

```diff
@@ -1,20 +1,73 @@
+import { useState, useRef } from 'react';
 import { ListItem } from '../components';
+import Button from '../components/Button';
 
 export function List({ data, listToken }) {
+	const [inputItem, setInputItem] = useState('');
+
+	// Initialize a useRef to bring focus back to input field
+	const filterInputRef = useRef(null);
+
+	// This fuction resets the input field:
+	const handleClear = () => {
+		setInputItem('');
+		filterInputRef.current.focus();
+	};
+
+	const handleInput = (e) => {
+		setInputItem(e.target.value);
+	};
+	// This helper function matches the user input any part of the item name from the items list
+	const stringMatch = (inputItem, listItem) => {
+		return listItem.toLowerCase().includes(inputItem.toLowerCase());
```

Comment:
**Praise:** Great work with this function - I agree with others that have said it was a great call to add `toLowerCase` here to make the filter a little smarter.

**Suggestion (non-blocking):** One way this function could be even further improved is to _sanitize_ the input coming from the user. If you're not familiar with the idea of sanitizing data, basically we want to prevent malicious users from trying to input harmful scripts into our app. Something we can do is use regex to [escape special characters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping).

```suggestion
		return listItem.toLowerCase().includes(inputItem.replace(/[.*+?^${}()|[\]\\]/g, '\\$&').toLowerCase());
```

(If you've never used regex before, this is a [helpful tool](https://regex101.com/) for getting more familiar with it!)

Reply:
Great suggestion Caitlin! We've included this improvement in our code, it's also a great review on regex expression